blockprod/src/detail/mod.rs

```diff
@@ -180,19 +180,31 @@ impl BlockProduction {
         &self,
         current_tip: Id<GenBlock>,
         min_block_timestamp: BlockTimestamp,
-    ) -> Result<Option<Box<dyn TransactionAccumulator>>, BlockProductionError> {
-        let max_block_size = self.chain_config.max_block_size_from_std_scripts();
+        transactions: Vec<SignedTransaction>,
+        transaction_ids: Vec<Id<Transaction>>,
+        packing_strategy: PackingStrategy,
+    ) -> Result<Box<dyn TransactionAccumulator>, BlockProductionError> {
+        let mut accumulator = Box::new(DefaultTxAccumulator::new(
+            self.chain_config.max_block_size_from_std_scripts(),
+            current_tip,
+            min_block_timestamp,
+        ));
+
+        for transaction in transactions.into_iter() {
+            let transaction_id = transaction.transaction().get_id();
+
+            accumulator
+                .add_tx(transaction, Amount::ZERO.into())
+                .map_err(|err| BlockProductionError::FailedToAddTransaction(transaction_id, err))?
```

Comment:
How are we making sure that these raw transactions are valid against what's coming from the mempool? I do hope we're not depending on the temporary solution we deployed that we wanted to solve later, where we include a transaction verifier inside the accumulator.

Reply:
There is no guarantee currently. Even transaction verifier is not sufficient because mempool deps are used to figure out the parent/child relationships. So it only properly works for transactions that are "independent" of transactions in the mempool. Could be removed since the guarantees are much weaker than for transactions specified by IDs.

Reply:
> I thought we agreed to just remove it...

Ok will do.

Reply:
Hmm there are many existing functional tests that rely on this functionality, and some are not easy to adapt so the transactions can be pushed through mempool.

Reply:
Let's just keep it, and write in the documentation that adding transactions there relies on the user ensuring the validity, and is done only in testing.